src/main/java/com/dnastack/wes/cromwell/CromwellService.java

```diff
@@ -211,13 +218,81 @@ private CromwellStatus getStatus(String runId) {
      *
      * @param runId The cromwell id
      *
-     * @return a complete run log
+     * @return the cromwell id
      */
     public RunId cancel(String runId) {
         client.abortWorkflow(runId);
         return RunId.builder().runId(runId).build();
     }
 
+    /**
+     * Get the files for a specific run.
+     *
+     * @param runId The cromwell id
+     *
+     * @return a list of generated files for the run
+     */
+    public RunFiles getRunFiles(String runId) throws NotFoundException {
+        CromwellMetadataResponse metadataResponse = getMetadata(runId);
+        Set<String> finalFileSet = new HashSet<>();
+        Set<String> secondaryFileSet = new HashSet<>();
+        Set<String> logFileSet = new HashSet<>();
+        List<RunFile> files = new ArrayList<>();
+
+        Map<String, Object> outputs = metadataResponse.getOutputs();
+        if (!outputs.isEmpty()) {
+            outputs.values().forEach(output -> extractFilesFromValue(finalFileSet, output));
+        }
+        extractSecondaryLogFiles(secondaryFileSet, logFileSet, metadataResponse);
+
+        finalFileSet.forEach(path -> files.add(new RunFile(RunFile.FileType.FINAL, path)));
+        secondaryFileSet.forEach(path -> {
+            if (!finalFileSet.contains(path) && !logFileSet.contains(path)) {
+                files.add(new RunFile(RunFile.FileType.SECONDARY, path));
+            }
+        });
+        logFileSet.forEach(path -> {
+            if (!finalFileSet.contains(path)) {
+                files.add(new RunFile(RunFile.FileType.LOG, path));
+            }
+        });
+        return new RunFiles(files);
+    }
+
+    /**
+     * Request to delete the files associated with the run.
+     *
+     * @param runId The cromwell id
+     *
+     * @return the cromwell id
+     */
+    public RunFileDeletions deleteRunFiles(String runId, boolean async) {
+        List<RunFile> files = getRunFiles(runId).runFiles();
+        List<RunFileDeletion> outcomes = files.stream().filter(runFile -> RunFile.FileType.SECONDARY.equals(runFile.getFileType()))
+            .map(runFile -> {
+                if (async) {
+                    return deleteRunFileAsync(runFile);
+                } else {
+                    return deleteRunFile(runFile);
+                }
+            }).toList();
+        return new RunFileDeletions(outcomes);
+    }
+
+    public RunFileDeletion deleteRunFileAsync(RunFile runFile) {
+        CompletableFuture.runAsync(() -> deleteRunFile(runFile), executor);
+        return new RunFileDeletion(runFile, RunFileDeletion.DeletionState.ASYNC,null);
+    }
+
+    public RunFileDeletion deleteRunFile(RunFile runFile) {
+        try {
+            storageClient.deleteFile(runFile.getPath());
```

Comment:
We may want to add a log message here:

```java
log.info("Deleting file '{}', runFile.getPath())
```
and then if there is an exception log an error
```java
log.error(Encountered exception while deleting file '%s': '%s'.formatted(runFile.getPath(),e.getMessage()),e)
```

Reply:
Added logging